### ISSUE TITLE
x/tools/present: initial comment line now interpretted as a title

### ISSUE THREAD

**Issue by kortschak** [golang/go#38578]
The present tool now supports Markdown syntax but also supports the previous present markup syntax. The switch to enable Markdown syntax use is that the first line (the talk title) be preceded with a `# `. Unfortunately, this coincides with the use of `# ` as a comment mark in the old syntax. In some cases, slides may contain copyright notices (I use present for lecture presentations and have done for some years now, so we require copyright notices). This puts the slide syntax into Markdown mode, which subsequently breaks because of the mutually unintelligible syntaxes.

It would be nice if there were a way to prevent present from assuming a `# ` prefix is a note to `present` to use Markdown. Or at the very least a more prominant warning that old slide have a significant risk of being broken (I'm actually happy we now have Markdown, but this will result in a fair bit of work for me to fix in my lecture presentations).

/cc @rsc

**Comment by lootch** [comment]
Hash-Bang rather than a pure Hash? I missed whatever discussion may
have occurred, so please ignore if that's already been rejected.

Lucio.


On 4/22/20, Dan Kortschak <notifications@github.com> wrote:
> The present tool now supports Markdown syntax but also supports the previous
> present markup syntax. The switch to enable Markdown syntax use is that the
> first line (the talk title) be preceded with a `# `. Unfortunately, this
> coincides with the use of `# ` as a comment mark in the old syntax. In some
> cases, slides may contain copyright notices (I use present for lecture
> presentations and have done for some years now, so we require copyright
> notices). This puts the slide syntax into Markdown mode, which subsequently
> breaks because of the mutually unintelligible syntaxes.
>
> It would be nice if there were a way to prevent present from assuming a `# `
> prefix is a note to `present` to use Markdown. Or at the very least a more
> prominant warning that old slide have a significant risk of being broken
> (I'm actually happy we now have Markdown, but this will result in a fair bit
> of work for me to fix in my lecture presentations).
>
> /cc @rsc
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/golang/go/issues/38578


-- 
Lucio De Re
2 Piet Retief St
Kestell (Eastern Free State)
9860 South Africa

Ph.: +27 71 471 3694
Cell: +27 83 251 5824

**Comment by kortschak** [comment]
Yeah, pretty much anything that's not `/# /` works. In the file I'm writing at the moment, I just did `s/^# /#_/`.

**Comment by andybons** [comment]
@rsc
